main.tf

```diff
@@ -2,41 +2,43 @@ resource "aws_vpc" "main" {
   cidr_block = "10.0.0.0/16"
 
 }
-resource "aws_subnet" "public_subnet1" {
+resource "aws_subnet" "public_subnet" {
+  count             = 2
```

Comment:
could you please change `count = 2` to `count = length(var.public_cidr)`  and also change for resources that has count  and is related to the public subnet